core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -129,14 +130,12 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
         if (queryBuilder != null) {
             searchSourceBuilder.query(queryBuilder);
         }
-
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        if (request.hasParam("from")) {
+            searchSourceBuilder.from(request.paramAsInt("from", DEFAULT_VALUE));
```

Comment:
`-1` was fine here. In this context it meant "this value is totally unused but I have to stick an integer in here because of the API design." `DEFAULT_VALUE` is bad because it makes the reader think that it is important because it is a constant but it isn't. Also I wouldn't name the default for two fields `DEFAULT_VALUE` that just causes all kinds of confusion. Changes like this can make reading the code way, way slower because you go off on mental tangents trying to figure out why `DEFAULT_VALUE` is named the way it is.